processing/src/test/java/org/apache/druid/segment/transform/TransformerTest.java

```diff
@@ -441,7 +445,74 @@ public void testTransformWithArrayDoubleInputs()
     Assert.assertEquals(2.3, (Double) raw[1], 0.0);
     Assert.assertNull(raw[2]);
     Assert.assertEquals(3.4, (Double) raw[3], 0.0);
-    Assert.assertEquals(Arrays.asList("1.2", "2.3", "null", "3.4"), actual.getDimension("dim"));
+    Assert.assertArrayEquals(new String[]{"1.2", "2.3", null, "3.4"}, actual.getDimension("dim").toArray());
     Assert.assertEquals(row.getTimestamp(), actual.getTimestamp());
   }
+
+  @Test
+  public void testTransformWithExpr()
+  {
+    final Transformer transformer = new Transformer(
+        new TransformSpec(
+            null,
+            ImmutableList.of(new ExpressionTransform("dim", "array_slice(dim, 0, 5)", TestExprMacroTable.INSTANCE))
+        )
+    );
+    final List<String> dimList = ImmutableList.of("a", "b", "c", "d", "e", "f", "g");
+    final MapBasedRow row = new MapBasedRow(
+        DateTimes.nowUtc(),
+        ImmutableMap.of("dim", dimList)
+    );
+    Assert.assertEquals(row.getDimension("dim"), dimList);
+    Assert.assertEquals(row.getRaw("dim"), dimList);
+
+    final InputRow actualTranformedRow = transformer.transform(new InputRow()
```

Comment:
I wonder if its necessary to use an inline `new InputRow()` here - do you need to retain the `row` in that type for some reason ?

you could make public/move outside/etc [this metod](https://github.com/apache/druid/blob/f0a70fe3c4533c889b6185774f104c0265a0c37e/indexing-hadoop/src/main/java/org/apache/druid/indexer/IndexGeneratorJob.java#L494)

...or probably try to use [MapBasedInputRow](https://github.com/apache/druid/blob/f0a70fe3c4533c889b6185774f104c0265a0c37e/processing/src/main/java/org/apache/druid/data/input/MapBasedInputRow.java#L38) ?

Reply:
yeah, I wanted to use the same flow that is being used in Hadoop indexer, thus created a inline InputRow in this test.